sklearn/metrics/_classification.py

```diff
@@ -401,7 +416,17 @@ def confusion_matrix(
             cm = cm / cm.sum(axis=0, keepdims=True)
         elif normalize == "all":
             cm = cm / cm.sum()
-        cm = np.nan_to_num(cm)
+
+        if xp.__name__ == "array_api_strict":
```

Comment:
Can we move this code as a helper inside the _array_api utils? You can find other examples there. Basically you can check if the xp namespace supports this function and simply call it otherwise you can use the code below to compute. This would also allow to remove an explicit check for "array_api_strict".

Reply:
Yes, I think that makes sense, since we use `np.nan_to_num()` in `BasesearchCV` and in `silhouette_samples()` as well.